### ISSUE TITLE
Unexpected `redundantReturn` behavior with `some View` functions

### ISSUE THREAD

**Issue by okalentiev** [nicklockwood/SwiftFormat#1819]
Swift compiler cannot infer return type without explicit `return`. 

```Swift
extension View {
    public func iOS16_scrollIndicatorsHidden() -> some View {
        if #available(iOS 16.0, *) {
            return self.scrollIndicators(.hidden)
        } else {
            return self
        }
    }
}
```

`SwiftFormat` removes returns from the above code, which makes compilation break because of the following error: 
```
Branches have mismatching types 'some View' and 'Self'
```

Looks like an additional use-case to https://github.com/nicklockwood/SwiftFormat/issues/1751 and https://github.com/nicklockwood/SwiftFormat/issues/1749

### Setup:
* `SwiftFormat` version: 0.54.3
* Both `conditionalAssignment` and `redundantReturn` are enabled
* `Swift` version 5.9

**Comment by nicklockwood** [comment]
@calda any thoughts on how we can detect this scenario?

**Comment by nicklockwood** [comment]
> Looks like an additional use-case to https://github.com/nicklockwood/SwiftFormat/issues/1751 and https://github.com/nicklockwood/SwiftFormat/issues/1749

I think these are actually all different scenarios, although they seem related

**Comment by calda** [comment]
This is a weird one. I guess it's because an if/switch expression has to return the same result type on each branch? In normal functions this is already enforced by the function having a single result type, which is not the case for a function with an opaque result type.

I think the fix would be to fixable the `redundantReturn` if/switch expression functionality within functions that return a `some` opaque result type.

**Comment by nicklockwood** [comment]
@okalentiev fixed in 0.54.4
